engine/src/main/java/org/hibernate/search/engine/search/highlight/dsl/CommonHighlightUnifiedOptionStep.java

```diff
@@ -0,0 +1,15 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.engine.search.highlight.dsl;
+
+/**
+ * TODO
+ */
+public interface CommonHighlightUnifiedOptionStep<S extends CommonHighlightUnifiedOptionStep<?>> extends CommonHighlightOptionStep<S> {
+	CommonHighlightBoundaryScannerTypeOptionStep<S> boundaryScanner();
```

Comment:
Hmm yeah ... I agree that `#5` does look best of these. `#2` is essentially the same as `#1` and with `#3-4` I feel like they are closer to that idea we were discussing initially to have the highlights similar to aggregations, and there's not much need to see the "select" part then...
As for "forcing" a user to pick a highlighter type at the very beginning, it was more to have an explicit type picked for future Lucene impl (as I'd assume that would lead to using different classes for different highlighter types), but we can actually just expose all the options from `CommonHighlightOptionStep`. Then on the other hand looking at the config options overall - I'd say that for users unfamiliar with Elasticsearch or Lucene most of those would look like some black magic 😄 
I'll give `#5` a try unless any other ideas will come up 😃

Reply:
> As for "forcing" a user to pick a highlighter type at the very beginning, it was more to have an explicit type picked for future Lucene impl (as I'd assume that would lead to using different classes for different highlighter types),

If you go for `#5`, it might actually be ok-ish to keep forcing users to select a highlighter type, since they will only be exposed to that if they start defining their highlighter and digging into advanced options.

I still think some sort of default would be nice, but we probably can start like this.